### ISSUE TITLE
使用Chord训练Qwen-25vl-7B使用zero3训练时出现RuntimeError: Expected all tensors to be on the same device, but found at least two devices, cuda:4 and cpu!

### ISSUE THREAD

**Issue by Strand2013** [modelscope/ms-swift#9521]
### Checklist / 检查清单

- [x] I have searched existing issues, and this is a new bug report. / 我已经搜索过现有的 issues，确认这是一个新的 bug report。

### Bug Description / Bug 描述

<img width="1028" height="404" alt="Image" src="https://github.com/user-attachments/assets/58b08d35-cfdc-4af1-a1fd-5a05e65c466d" />

A800
NVIDIA-SMI 550.144.04
Driver Version: 550.144.04
CUDA Version: 13.0
torch    2.11.0+cu130
torchaudio    2.11.0+cu130
torchvision    0.26.0+cu130
ms_swift   4.2.0

### How to Reproduce / 如何复现

训练脚本：

swift rlhf \
    --rlhf_type grpo \
    --template qwen2_5_vl \
    --model ${ACTOR_PATH} \
    --dataset ${DATASET_PATH} \
    --load_from_cache_file true \
    --torch_dtype bfloat16 \
    --beta 0.0 \
    --num_train_epochs $max_epoch \
    --add_non_thinking_prefix true \
    --split_dataset_ratio 0.01 \
    --per_device_train_batch_size $bsz \
    --per_device_eval_batch_size 1 \
    --gradient_accumulation_steps 8 \
    --chord_sft_per_device_train_batch_size 1 \
    --chord_sft_dataset ${DATASET_PATH} \
    --chord_enable_phi_function false \
    --chord_mu_warmup_steps 0 \
    --chord_mu_decay_steps 200 \
    --chord_mu_peak 0.9 \
    --chord_mu_valley 0.05 \
    --num_generations 8 \
    --tuner_type lora \
    --lora_rank 16 \
    --lora_alpha 32 \
    --lorap_lr_ratio 10 \
    --freeze_vit false \
    --freeze_aligner false \
    --freeze_llm false \
    --target_modules qkv all-linear \
    --reward_funcs accuracy \
    --use_vllm true \
    --vllm_mode colocate \
    --vllm_gpu_memory_utilization 0.4 \
    --vllm_max_model_len 8192 \
    --max_completion_length 4096 \
    --overlong_filter true \
    --offload_optimizer true \
    --offload_model true \
    --sleep_level 1 \
    --weight_decay 0.1 \
    --group_by_length false \
    --learning_rate 1e-6 \
    --warmup_ratio 0.05 \
    --lazy_tokenize true \
    --dataloader_num_workers 0 \
    --dataset_num_proc 8 \
    --gradient_checkpointing true \
    --attn_impl flash_attn \
    --eval_steps 99000000 \
    --save_steps 100 \
    --save_total_limit 5 \
    --logging_steps 10 \
    --max_length 4096 \
    --output_dir ${PRIMUS_SAVE_CHECKPOINT_DIR} \
    --logging_dir ${PRIMUS_TENSORBOARD_LOG_DIR} \
    --add_version False \
    --deepspeed zero3 \
    --log_completions true \
    --report_to tensorboard

**Comment by hjh0119** [comment]
should be fixed in #9525
